converter/sfxConverter.go

```diff
@@ -452,16 +469,170 @@ func NewCadvisorCollector(infoProvider infoProvider, f ContainerNameToLabelsFunc
+				extraLabels: []string{},
+			},
+			getValues: func(container *info.ContainerInfo) metricValues {
+				return metricValues{{value: datapoint.NewIntValue(int64(container.Spec.Cpu.Limit))}}
+			},
+		},
+	}
+
+	var index = 0
+	for _, metric := range metrics {
+		// check if metric is not on the exclusion list
+		if _, ok := excludedMetrics[metric.name]; !ok {
+			metrics[index] = metric
+			index++
+		}
+	}
+	// trim metrics down to the desired length
+	metrics = metrics[:index]
+	return metrics
+}
+
+func getContainerSpecMemMetrics(excludedMetrics map[string]bool) []containerSpecMetric {
+	var metrics = []containerSpecMetric{
+		{
+			containerMetric: containerMetric{
+				name:        "container_spec_memory_limit_bytes",
+				help:        "",
+				valueType:   datapoint.Gauge,
+				extraLabels: []string{},
+			},
+			getValues: func(container *info.ContainerInfo) metricValues {
+				return metricValues{{value: datapoint.NewIntValue(int64(container.Spec.Memory.Limit))}}
+			},
+		},
+		{
+			containerMetric: containerMetric{
+				name:        "container_spec_memory_swap_limit_bytes",
+				help:        "",
+				valueType:   datapoint.Gauge,
+				extraLabels: []string{},
+			},
+			getValues: func(container *info.ContainerInfo) metricValues {
+				return metricValues{{value: datapoint.NewIntValue(int64(container.Spec.Memory.SwapLimit))}}
+			},
+		},
+	}
+
+	var index = 0
+	for _, metric := range metrics {
+		// check if metric is not on the exclusion list
+		if _, ok := excludedMetrics[metric.name]; !ok {
+			metrics[index] = metric
+			index++
+		}
+	}
+	// trim metrics down to the desired length
+	metrics = metrics[:index]
+	return metrics
+}
+
+func getMachineInfoMetrics(excludedMetrics map[string]bool) []machineInfoMetric {
+	var metrics = []machineInfoMetric{
+		{
+			containerMetric: containerMetric{
+				name:        "machine_cpu_frequency_khz",
+				help:        "",
+				valueType:   datapoint.Gauge,
+				extraLabels: []string{},
+			},
+			getValues: func(machineInfo *info.MachineInfo) metricValues {
+				return metricValues{{value: datapoint.NewIntValue(int64(machineInfo.CpuFrequency))}}
+			},
+		},
+		{
+			containerMetric: containerMetric{
+				name:        "machine_cpu_cores",
+				help:        "",
+				valueType:   datapoint.Gauge,
+				extraLabels: []string{},
+			},
+			getValues: func(machineInfo *info.MachineInfo) metricValues {
+				return metricValues{{value: datapoint.NewIntValue(int64(machineInfo.NumCores))}}
+			},
+		},
+		{
+			containerMetric: containerMetric{
+				name:        "machine_memory_bytes",
+				help:        "",
+				valueType:   datapoint.Gauge,
+				extraLabels: []string{},
+			},
+			getValues: func(machineInfo *info.MachineInfo) metricValues {
+				return metricValues{{value: datapoint.NewIntValue(int64(machineInfo.MemoryCapacity))}}
+			},
+		},
+	}
+	var index = 0
+	for _, metric := range metrics {
+		// check if metric is not on the exclusion list
+		if _, ok := excludedMetrics[metric.name]; !ok {
```

Comment:
If the metric is excluded we we don't reassign it to the current position and  we don't increment index.  Then if the next metric isn't excluded we set the current index to that metric and increment the index.  This way we put all "included" metrics in the front of the array 0...index.  Then we slice the array on index.